### PR TITLE
Code consistency and Ruby 2.7 Support

### PR DESCRIPTION
**Pull request by stevenharman**
## Notable Changes
* Fix Ruby 2.7 warnings
* Drop old/broken rubocop config and format/lint with standardrb
* Drop EoL'd Rubies
* Add CHANGELOG.md

### CODE REVIEW

**Comment by rwz** [comment]
Could you give me until tomorrow to look through the changes?

File: l2meter.gemspec [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+    "changelog_uri" => "#{spec.homepage}/blob/master/CHANGELOG.md"
+  }
+
+  # Specify which files should be added to the gem when it is released.
+  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
+  spec.files = Dir.chdir(File.expand_path("..", __FILE__)) do
+    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
+  end
+  spec.bindir = "exe"
+  spec.executables = []
```
Comment (rwz) [inline l2meter.gemspec:27]: This project doesn't have binaries or executables, so why include this?

File: lib/l2meter/emitter.rb [hunk 2]
```diff
@@ -1,5 +1,4 @@
 require "time"
-require "thread"
```
Comment (rwz) [inline lib/l2meter/emitter.rb:2]: You don't need to require thread to get Mutex defined anymore? Is this a change in newer ruby versions?

File: spec/emitter_spec.rb [hunk 3]
```diff
@@ -590,35 +588,33 @@
...
+        /secret/.match?(value) ? nil : value
       end
 
-      subject.log \
-        hello: :world,
-        my_password: "this secret schould be scrubbed",
-        other_field: -> { "also secret, should be scrubbed" }
+      subject.log(hello: :world,
+                  my_password: "this secret schould be scrubbed",
+                  other_field: -> { "also secret, should be scrubbed" })
```
Comment (rwz) [inline spec/emitter_spec.rb:617]: Omg, does standardrb really suggests this awful syntax?

File: l2meter.gemspec [hunk 4]
```diff
@@ -1,15 +1,31 @@
...
+    "source_code_uri" => spec.homepage,
+    "bug_tracker_uri" => "#{spec.homepage}/issues",
+    "changelog_uri" => "#{spec.homepage}/blob/master/CHANGELOG.md"
+  }
+
+  # Specify which files should be added to the gem when it is released.
+  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
+  spec.files = Dir.chdir(File.expand_path("..", __FILE__)) do
+    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
+  end
```
[the remark this answers is not in the document]
Comment (stevenharman) [reply l2meter.gemspec:25]: That's fair. And I'm not in love with shelling out either. One thing I don't like about the static list is that it rots. e.g., I added the `CHANGELOG.md` and forgot to add it at first. I only knew to add it b/c I also happened to be cleaning up the `*.gemspec` at the same time.

On other point: this is what a `bundle gem` generates. 🤷

All of that said, [I'm like a 3/10 on this decision](https://capwatkins.com/blog/the-sliding-scale-of-giving-a-fuck). What about you?

File: l2meter.gemspec (hunk 1 above)
Comment (stevenharman) [reply l2meter.gemspec:27]: To be explicit, mostly.

File: lib/l2meter/emitter.rb (hunk 2 above)
Comment (stevenharman) [reply lib/l2meter/emitter.rb:2]: As of Ruby 2.2-ish? Anyhow, it's one of the "safe auto fixes" that `standard` does for us.

File: spec/emitter_spec.rb (hunk 3 above)
Comment (stevenharman) [reply spec/emitter_spec.rb:617]: Not exactly. This had some manual intervention. The automated formatting actually left the `\`, even with the parens. Hopefully the forthcoming `rubyfmt` will have a better answer there.

Anyhow, I removed the then redundant `\` and pulled the lines up. At that point the auto-formatting resulted in what we see here. If we push the first line back down, the auto-formatting will result in this:

```ruby
subject.log(
  hello: :world,
  my_password: "this secret schould be scrubbed",
  other_field: -> { "also secret, should be scrubbed" }
)```

Which I can totally live with. Maybe we should go that route so it's less churn?

Comment (stevenharman) [reply spec/emitter_spec.rb]: **UPDATE**: I rebased in a change to keep the original formatting, but with parens. 😄

Comment (rwz) [reply spec/emitter_spec.rb]: I like it way more, thanks!

**Pull request merged**